Assets/Scripts/Player/PlayerControls.cs

```diff
@@ -48,7 +51,12 @@ void Update()
 
         if (Input.GetMouseButton(0))
         {
-            ShootControl.Shoot(_gun);
+            if (Time.time >= last_shot + stats.fireRate)
+            {
+                last_shot = Time.time;
+                ShootControl.Shoot(_gun);
+                stats.Ammo -= 1;
+            }
```

Comment:
I'm already working on this part so you can just revert back to 
```suggestion
                ShootControl.Shoot(_gun);
``` 
The firerate is being incorporated in a Weapon class